src/OpenSage.Game/Data/Wav/WavFile.cs

```diff
@@ -65,51 +135,118 @@ public sealed class WavFile
     {
         private RiffHeader _header;
         private WaveFormat _format;
+        private WaveFact _fact;
         private WaveData _data;
         private byte[] _buffer;
 
-        public int Size =>  (int)_data.SubChunkSize;
-        public int Fequency => (int)_format.SampleRate;
+        public int Size => _buffer.Length;
+        public int Frequency => (int)_format.SampleRate;
+        public int AudioFormat => _format.AudioFormat;
         public int Channels => _format.NumChannels;
         public int BitsPerSample => _format.BitsPerSample;
         public byte[] Buffer => _buffer;
 
-        public void Parse(BinaryReader reader)
+        private int ByteToInt16(byte[] packed)
         {
-            _header = RiffHeader.Parse(reader);
+            // This is always little endian, unlike the C# builtin for unpacking a byte array.
+            return (packed[1] << 8) | packed[0];
+        }
 
-            if (new string(_header.ChunkId) != "RIFF" ||
-                new string(_header.Format) != "WAVE")
+        private byte[] ParseMicrosoft(BinaryReader reader, int size)
```

Comment:
It might be cleaner to implement a WavParser baseclass and add two derived classes of type PcmParser and AdpcmParser. I am pretty sure i saw other compressed .wav formats in BFME before, so it should be easier in the future to add new formats with that approach. But i can do that myself after merge if you don't want that refactoring now.